internal/broker/config.go

```diff
@@ -86,11 +176,48 @@ func parseConfigFile(cfgPath string) (userConfig, error) {
 		cfg.clientSecret = oidc.Key(clientSecret).String()
 	}
 
-	users := iniCfg.Section(usersSection)
-	if users != nil {
-		cfg.homeBaseDir = users.Key(homeDirKey).String()
-		cfg.allowedSSHSuffixes = strings.Split(users.Key(sshSuffixesKey).String(), ",")
-	}
+	populateUsersConfig(&cfg, iniCfg.Section(usersSection), p)
 
 	return cfg, nil
 }
+
+func (uc *userConfig) shouldRegisterOwner() bool {
+	return uc.ownerAllowed && uc.firstUserBecomesOwner && uc.owner == ""
+}
+
+func (uc *userConfig) registerOwner(cfgPath, userName string) error {
+	if cfgPath == "" {
+		uc.owner = userName
+		uc.firstUserBecomesOwner = false
+		return nil
+	}
+
+	p := filepath.Join(GetDropInDir(cfgPath), ownerAutoRegistrationConfigPath)
+
+	templateName := filepath.Base(ownerAutoRegistrationConfigTemplate)
+	t, err := template.New(templateName).ParseFS(ownerAutoRegistrationConfig, ownerAutoRegistrationConfigTemplate)
+	if err != nil {
+		slog.Error(fmt.Sprintf("Failed to open autoregistration template: %v", err))
+		return err
```

Comment:
I feel it would be better to just:

```suggestion
		return fmt.Errorf("failed to open autoregistration template: %w", err)
```

Then in the caller function to log in case of error.

Reply:
agreed